src/models/data.cpp

```diff
@@ -353,3 +353,52 @@ const Data::ThreadEvents* Data::EventResults::findThread(qint32 pid, qint32 tid)
 {
     return const_cast<Data::EventResults*>(this)->findThread(pid, tid);
 }
+
+BottomUpResults BottomUpResults::mergeBottomUpResults(const BottomUpResults &a, const BottomUpResults &b)
+{
+    if (a.costs.numTypes() != b.costs.numTypes()) {
+        return {};
+    }
+
+    BottomUpResults results;
+
+    for (int i = 0; i < a.costs.numTypes(); i++) {
+        results.costs.addType(i * 2, a.costs.typeName(i) + QStringLiteral(" - file A"), a.costs.unit(i));
+        results.costs.addTotalCost(i * 2, a.costs.totalCost(i));
+    }
+
+    for (int i = 0; i < b.costs.numTypes(); i++) {
+        results.costs.addType(i * 2 + 1, b.costs.typeName(i) + QStringLiteral(" - file B"), b.costs.unit(i));
+        results.costs.addTotalCost(i * 2 + 1, b.costs.totalCost(i));
+    }
+
+    std::function<void(const BottomUp&, BottomUp& parent)> iterateModels = [&iterateModels, a, b, &results](const BottomUp& node, BottomUp& parent) {
+        BottomUp bottomUp;
+        bottomUp.id = node.id;
+        bottomUp.symbol = node.symbol;
+
+        for (int i = 0; i < a.costs.numTypes(); i++) {
+            results.costs.add(i * 2, bottomUp.id, a.costs.cost(i, bottomUp.id));
+        }
+
+        const auto sibling = b.root.entryForSymbol(node.symbol);
```

Comment:
this looks wrong to me - you'll only lookup on the root level, but instead you should actually go deeper in both, `a` and `b` simultaneously and then look for the matching entry at the corresponding levels, no?